artcommon/artcommonlib/assembly.py

```diff
@@ -269,7 +270,7 @@ def _assembly_field(field_name: str, releases_config: Model, assembly: str) -> M
     return Model(dict_to_model=config_dict)
 
 
-def assembly_basis_event(releases_config: Model, assembly: str, strict: bool = False) -> typing.Optional[int]:
+def assembly_basis_event(releases_config: Model, assembly: str, strict: bool = False) -> typing.Optional[typing.Union[int, datetime]]:
```

Comment:
hmm why not a new method for this, so we know for sure which field is referenced and what type will be returned ?

Reply:
Why bother? We've always chosen to make code flexible instead of duplicating logic across build systems